src/coreclr/src/vm/olevariant.cpp

```diff
@@ -400,14 +403,8 @@ VARTYPE OleVariant::GetVarTypeForTypeHandle(TypeHandle type)
         return VT_UNKNOWN;
     }
 
-    // Handle array's.
-    if (!CorTypeInfo::IsArray(elemType))
-    {
-        // Non interop compatible type.
-        COMPlusThrow(kArgumentException, IDS_EE_COM_UNSUPPORTED_SIG);
-    }
-
-    return VT_ARRAY;
+    // Non interop compatible type.
```

Comment:
Nit: It may be nice to move this up to make the code more linear:

```
if (type.IsTypeDesc())
    COMPlusThrow(kArgumentException, IDS_EE_COM_UNSUPPORTED_SIG);
```